### PR TITLE
Migrate to Github Actions

### LINKED ISSUE #70: Very long TravisCI builds
### Problem

I recently submitted a couple of pull requests to fix some issues I was having, and noticed that the TravisCI build was very slow, sometimes taking several hours to just start and a couple more to finish.

This seems caused by Travis [recent pricing change](https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing) which limit open source to 1000 minutes, I suppose jazzband is hitting this quota very quickly.

Since this is not an issue specific to `django-redshift-backend`, I first tried to reach out on IRC, but I'm not sure if my message went through....

#### Procedure to reproduce the problem

Example of pull request: https://github.com/jazzband/django-redshift-backend/pull/68
Build for the first commit: https://travis-ci.org/github/jazzband/django-redshift-backend/builds/741443844

PR submitted on 4th Nov 5:06pm GMT
Build started 5th Nov 1:32am GMT
Build finished 5th Nov 3:30am GMT

#### Suggested Solution

I think it's easy to agree that this is not sustainable for contributors and even less for maintainers. I'd be happy to help, but first I'd like to confirm the CI that the Jazzband org wants to use. I assume Github actions is the easiest these days as it doesn't require maintainers to grant anything and it has a generous free allowance.

The only part I'm seeing where we would need a maintainer to step in is to add the PyPI token to the repository secrets to keep release automation working.

FYI @jezdez as roadie and @shimizukawa as recent active maintainer...

### PR DESCRIPTION
**Pull request by browniebroke**
Subject: Move CI to Github as Travis builds have become too slow

### Feature or Bugfix
- Feature

### Purpose
Move the CI to Github Actions. The config is a bit more verbose than I'd like to, but this should have the advantage to be explicit.

### Detail
- Fixes #70

### Relates
I have another PR that will conflict: #68 

Given that Travis is too painful to work with, I'll update that other PR when this is merged.

### :warning: IMPORTANT :warning: 

**A project admin will have to add `JAZZBAND_RELEASE_KEY` to the repo secrets: https://github.com/jazzband/django-redshift-backend/settings/secrets/actions/new**

### CODE REVIEW

File: django_redshift_backend/base.py
Comment (browniebroke) [inline django_redshift_backend/base.py:97]: This [was failing flake8](https://github.com/browniebroke/django-redshift-backend/runs/1364132287?check_suite_focus=true#step:7:13).

File: .github/workflows/ci.yml
Comment (browniebroke) [inline .github/workflows/ci.yml:91]: Note that the deploy step used to run on Python 2.7.

**Comment by browniebroke** [comment]
Not sure why the builds aren't running, but they do run on my fork: https://github.com/browniebroke/django-redshift-backend/actions?query=workflow%3ACI

File: .github/workflows/ci.yml
Comment (shimizukawa) [inline .github/workflows/ci.yml:99]: I don't know about the Jazzband way of releasing, but I think `pypa/gh-action-pypi-publish` with the API provided by PyPI would be better in the future.

File: .github/workflows/ci.yml
Comment (shimizukawa) [reply .github/workflows/ci.yml:89]: Since `django-redshift-backend` doesn't create a binary wheel, so I don't think that's a problem.

File: .github/workflows/ci.yml
Comment (shimizukawa) [inline .github/workflows/ci.yml:78]: I've seen the results of your execution. I think it's perfect 🎉 
https://github.com/browniebroke/django-redshift-backend/runs/1364264120

**Review by shimizukawa — APPROVED** [review]
LGTM!
Thanks!!

File: .github/workflows/ci.yml
Comment (shimizukawa) [reply .github/workflows/ci.yml:99]: @jezdez please check this deploy section. After you agree on the deploy section, we can merge this PR.

Comment (jezdez) [reply .github/workflows/ci.yml:99]: This isn't using the Jazzband package index, so will need to update to use it. I'll get to do a proper review later tonight. Don't merge this before.

**Comment by jezdez** [comment]
Please use [tox-gh-actions](https://github.com/ymyzk/tox-gh-actions) for configuring the test matrix instead of duplicating it in the GH actions and tox configuration. This is along the lines we've done with tox-travis.

File: .github/workflows/ci.yml
Comment (browniebroke) [reply .github/workflows/ci.yml:72]: I looked at [Jazzband's release process](https://jazzband.co/about/releases) and I think I get it... Packages are not published to PyPI directly, they are first going to Jazzband package index which in turns publich to PyPI? I'll copy [the publish part from `prettytable`](https://github.com/jazzband/prettytable/blob/f9913bb9d3898c86b730387d297dca44a647756e/.github/workflows/deploy.yml#L44-L51), should be a good starting point.

**Comment by browniebroke** [comment]
Updated test to use `tox-gh-actions` and deployment steps to use Jazzband package index.

Build result: https://github.com/browniebroke/django-redshift-backend/actions/runs/352344702

File: .github/workflows/ci.yml
Comment (jezdez) [inline .github/workflows/ci.yml:84]: This needs to be updated:

```suggestion
          repository_url: https://jazzband.co/projects/django-redshift-backend/upload
```

**Review by jezdez — CHANGES_REQUESTED** [review]
Thanks, this is looking more and more like what we need, thank you for working on this!

A few things that would be better:
- separate [test](https://github.com/jazzband/prettytable/blob/master/.github/workflows/test.yml), [lint](https://github.com/jazzband/prettytable/blob/master/.github/workflows/lint.yml) and [deploy](https://github.com/jazzband/prettytable/blob/master/.github/workflows/deploy.yml) workflows like in prettytable
- the deploy workflow should use the "published" event type instead of checking manually for the existence of a git tag

**Comment by browniebroke** [comment]
I _think_ it's all done. Let me know if I missed anything.

**Comment by browniebroke** [comment]
Is there anything else I need to change here?

**Comment by shimizukawa** [comment]
@browniebroke Thank you for your work!

@jezdez I'm going to release a new version after merging this PR. What do we need next?

**Comment by shimizukawa** [comment]
@jezdez ping and Happy New Year!!

**Comment by jezdez** [comment]
I've fixed a few more things in #72. Closing this PR since I wasn't able to push to the  branch of this PR being in a fork repo.

**Comment by shimizukawa** [comment]
Now I've released 2.0.0. Thanks for your contribution!
https://pypi.org/project/django-redshift-backend/2.0.0/

**Pull request closed without merging**
